### PR TITLE
dispatcher component

### PR DESCRIPTION
**Pull request by fdegenhr**

### CODE REVIEW

**Comment by philecs** [comment]
@fdegenhr what about the resource as a needed characteristic? Maybe a tenant only wants an application to operate within a specific domain?

Example: `Resource`.Operations : `Contact`.Create

File: docs/component-dispatcher.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+The dispatcher service is a central component in the hub and spoke service collaboration. 
+It is responsible for delivering messages from one application to one or more target applications based on a tenant specific configuration.
+
+![Dispatcher Service Overview](https://github.com/openintegrationhub/openintegrationhub/blob/dispatcher/Assets/component_dispatcher_0.1.png)
+
+## Service
+### Entry Point
+
+The service is triggered by messages arriving through a SDF Adapter Component used in a Connector Flow. 
+
```
Comment (philecs) [inline docs/component-dispatcher.md:12]: Is the data really received by sdf adapter? 
Is it possible that the dispatcher component receives data from another service? (E.g. data was validated by some sort of validation service and is then passed to the dispatcher component)

File: docs/component-dispatcher.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+| cms1 | product | true | false | true |
+
+
+
+For each received message, the configuration is evaluated to determin how to process the message. 
+Message characteristics relevant for the dispatcher service:
+* tenant id
+* source application
+* source operation
+
```
Comment (philecs) [inline docs/component-dispatcher.md:37]: What about the resource as a needed characteristic? Maybe a tenant only wants an application to operate within a specific domain?

Example: `Resource`.Operations : `Contact`.Create

File: docs/component-dispatcher.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+After evaluating the tenant specific configuration for a message, the configured target applications are known.
+
+### message processing
+
+If a tenant specified one or more connected application which should receive the message, the original message is duplicated for each target application. Those messages will be delivered to the target applications.
+Each of those messages has new message id. The transaction id will remain the same.
```
Comment (philecs) [inline docs/component-dispatcher.md]: Is it necessary to duplicate the message if there is only one specified application?

File: docs/component-dispatcher.md (hunk 2 above)
Comment (fdegenhr) [reply docs/component-dispatcher.md:37]: source domain added

File: docs/component-dispatcher.md (hunk 3 above)
Comment (fdegenhr) [reply docs/component-dispatcher.md]: rephrased

**Review by philecs — APPROVED** [review]

**Pull request merged**